src/gui/inputdevices/old/InputSequence.cpp

```diff
@@ -391,30 +393,33 @@ void InputSequence::stopInput() {
  */
 auto InputSequence::changeTool() -> bool {
     Settings* settings = inputHandler->getSettings();
-    ButtonConfig* cfgTouch = settings->getTouchButtonConfig();
+    ButtonConfig* cfgTouch = settings->getButtonConfig(Buttons::BUTTON_TOUCH);
     ToolHandler* h = inputHandler->getToolHandler();
     GtkXournal* xournal = inputHandler->getXournal();
+    bool toolChanged = false;
+    bool pressed = false;
+    bool configChanged = true;
 
-    ButtonConfig* cfg = nullptr;
     if (gdk_device_get_source(device) == GDK_SOURCE_PEN) {
         penDevice = true;
-        if (button == 2) {
-            cfg = settings->getStylusButton1Config();
-        } else if (button == 3) {
-            cfg = settings->getStylusButton2Config();
-        }
+        if (button == 2)
+            InputUtils::applyButton(h, settings, Buttons::BUTTON_STYLUS, pressed, toolChanged, configChanged);
+        else if (button == 3)
+            InputUtils::applyButton(h, settings, Buttons::BUTTON_STYLUS2, pressed, toolChanged, configChanged);
     } else if (button == 2 /* Middle Button */ && !xournal->selection) {
-        cfg = settings->getMiddleButtonConfig();
+        InputUtils::applyButton(h, settings, Buttons::BUTTON_MIDDLE, pressed, toolChanged, configChanged);
     } else if (button == 3 /* Right Button */ && !xournal->selection) {
-        cfg = settings->getRightButtonConfig();
+        InputUtils::applyButton(h, settings, Buttons::BUTTON_RIGHT, pressed, toolChanged, configChanged);
     } else if (gdk_device_get_source(device) == GDK_SOURCE_ERASER) {
         penDevice = true;
-        cfg = settings->getEraserButtonConfig();
+        InputUtils::applyButton(h, settings, Buttons::BUTTON_ERASER, pressed, toolChanged, configChanged);
     } else if (cfgTouch->device == gdk_device_get_name(device)) {
-        cfg = cfgTouch;
+        pressed = true;
```

Comment:
Very obscure way of calling some specific methods later on. Just call them directly here.

Reply:
You can actually ignore this as it is part of the old input system which is listed for removal. See #2308

Reply:
Thinking of that - it might be best if we remove the old input system first and then merge your PR.

Reply:
In PR #2308 @Technius wrote:

>  I think this should be merged after we release a beta/milestone version of current master so that the transition is not so sudden.

I don't think this PR (#2339) should wait until a beta/milestone version of current master is released. So you two @Technius @LittleHuba should agree on when the old input system gets removed.

Reply:
Then this one first as it is not a breaking change but a fix. But #2308 should follow with small delay as this PR most likely breaks the old system even more.

Reply:
Just to be sure this is not missed: 
The addition of the fix for "Drawing when trying to unselect" (#1797) that I added in one of the last commits could be experienced as breaking change by some users even though in my opinion it's rather a fix to a usability issue. Or at least it is for my usage of xournalpp.
I would still keep it in but in case you would rather want to have it separately I can also revert that fix.

Reply:
For me this is also a fix rather than a change. We added other fixes you could see as "breaking" changes between earlier minor versions so no issue here.

Reply:
Please resolve if you think this is fine :) (The cleanup happened in38d509654dd5b017f84bf8b17f59b0a907d83638)